### ISSUE TITLE
[C#] Implement new Concept Exercise: namespaces

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/v3#1127]
This issue describes how to implement the `namespaces` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the Concept of Namespaces in C#.

## Learning objectives

- Know what namespaces are.
- Know how to import namespaces.

## Out of scope

- Assemblies.

## Concepts

This Concepts Exercise's Concepts are:

- `namespaces`: know what namespaces are; know how to import namespaces.
- `accessibility`: know how to use access modifiers to limit access to elements.

## Prequisites

This Concept Exercise's prerequisites Concepts are:

- `basics`: know how to define a static class and static method.
- `classes`: know how to define a class and what the `public` and `private` access modifiers are.
- `inheritance`: know about the `protected` access modifier.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Namespaces][namespaces]: how to define and import namespaces.
- [Accessibility levels][accessibility-levels]: use the `public/internal/private` access modifiers.

### After

- [Namespaces][namespaces]: how to define and import namespaces.
- [Accessibility levels][accessibility-levels]: use the `public/internal/private` access modifiers.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/README.md#reference-docs
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer
[exercise-example]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/numbers-floating-point
[design-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/design.md
[config.json-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/config.json
[concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[namespaces]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/namespaces/
[accessibility-levels]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/accessibility-levels

**Comment by mikedamay** [comment]
Perhaps we could start with a nested type situation and then refactor that to use a namespace instead.

**Comment by ErikSchierboom** [comment]
Yep, that could work!

**Comment by mikedamay** [comment]
Closed with PR  #1987
